### PR TITLE
Typescript charts

### PR DESCRIPTION
**Pull request by SanjeevLakhwani**

### CODE REVIEW

File: src/js/components/Overview/charts/chartTypes.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  name: string;
+  payload: {
+    x: string;
+  };
+  value: number;
+}
+
+export type HexColor = `#${string}`;
+
+export type ChartTheme = {
```
Comment (v-rocheleau) [inline src/js/components/Overview/charts/chartTypes.ts:20]: Neat, we could use generics here but I dont think its necessary at this point

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
+  "compilerOptions": {
+    "outDir": "./dist/",
```
Comment (v-rocheleau) [inline tsconfig.json:3]: The build directory should be added to the .gitignore

File: webpack.config.js [hunk 3]
```diff
@@ -3,16 +3,17 @@ const HtmlWebpackPlugin = require('html-webpack-plugin');
 const CopyWebpackPlugin = require('copy-webpack-plugin');
 
 const config = {
+  mode: 'development',
```
Comment (v-rocheleau) [inline webpack.config.js:6]: The mode should be determined by the deployment environment variables I think.

**Review by v-rocheleau — CHANGES_REQUESTED** [review]
Looks good! Added comments on little details

File: src/js/components/Overview/Chart.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { chartTypes } from '../../constants/overviewConstants';
+import BentoBarChart from './charts/BentoBarChart';
+import BentoPie from './charts/BentoPie';
+
+import { CHART_HEIGHT } from '../../constants/overviewConstants';
+
+const Chart = ({ chartType, data, units }) => {
```
[the remark this answers is not in the document]
Comment (SanjeevLakhwani) [reply src/js/components/Overview/Chart.js:9]: Not in the scope for this PR

File: src/js/components/Overview/charts/BentoPie.tsx [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+  UnitaryMapCallback,
+} from './chartTypes';
+import { useChartTheme, useChartTranslation } from './ChartConfigProvider';
+import { polarToCartesian } from './chartUtils';
+
+const chartAspectRatio = 1.4;
+const LABEL_THRESHOLD = 0.05;
+const MAX_LABEL_CHARS = 14;
+
+const textStyle: CSS.Properties = {
```
[the remark this answers is not in the document]
Comment (SanjeevLakhwani) [reply src/js/components/Overview/charts/BentoPie.tsx:37]: done

File: src/js/components/Overview/charts/BentoBarChart.tsx [hunk 6]
```diff
@@ -19,34 +21,51 @@ const TICKS_SHOW_ALL_LABELS_BELOW = 11; // Below this # of X-axis ticks, force-s
...
     return tickLabel;
   }
   return `${tickLabel.substring(0, MAX_TICK_LABEL_CHARS)}...`;
 };
-const fill = (entry) => (entry.x === 'missing' ? CHART_MISSING_FILL : BAR_CHART_FILL);
 
-const BentoBarChart = ({ title, data, units, height }) => {
-  const { t } = useTranslation(NON_DEFAULT_TRANSLATION);
-  const { t: td } = useTranslation(DEFAULT_TRANSLATION);
+const BentoBarChart = ({
```
[the remark this answers is not in the document]
Comment (SanjeevLakhwani) [reply src/js/components/Overview/charts/BentoBarChart.tsx:26]: done

File: src/js/components/Overview/charts/BentoPie.tsx [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+const BentoPie = ({
+  data,
+  height,
+  sort = true,
+  preFilter,
+  dataMap,
+  postFilter,
+  removeEmpty = true,
+  colorTheme = 'default',
+}: {
```
[the remark this answers is not in the document]
Comment (SanjeevLakhwani) [reply src/js/components/Overview/charts/BentoPie.tsx]: done

File: webpack.config.js (hunk 3 above)
Comment (SanjeevLakhwani) [reply webpack.config.js:6]: It is being defined by scripts this is just for cases if the user runs a specific command. (which would only happen if they are in dev)

File: src/js/components/Overview/charts/ChartConfigProvider.tsx [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  theme?: ChartTheme;
+  Lng: string;
+  translationMap?: TranslationObject;
+  children: React.ReactElement;
+}) => {
+  let lang: SupportedLng = 'en';
+  try {
+    lang = Lng as SupportedLng;
+  } catch (e) {
+    console.error('Lng is not a supported language');
```
Comment (davidlougheed) [inline src/js/components/Overview/charts/ChartConfigProvider.tsx:27]: turn this into a more useful error message by including `${lang}` in there

**Review by v-rocheleau — APPROVED** [review]

**Pull request merged**